app/src/main/java/com/keylesspalace/tusky/components/search/fragments/SearchFragment.kt

```diff
@@ -0,0 +1,128 @@
+package com.keylesspalace.tusky.components.search.fragments
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProviders
+import androidx.paging.PagedList
+import androidx.paging.PagedListAdapter
+import androidx.recyclerview.widget.DividerItemDecoration
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.google.android.material.snackbar.Snackbar
+import com.keylesspalace.tusky.AccountActivity
+import com.keylesspalace.tusky.BottomSheetActivity
+import com.keylesspalace.tusky.R
+import com.keylesspalace.tusky.ViewTagActivity
+import com.keylesspalace.tusky.components.search.SearchViewModel
+import com.keylesspalace.tusky.di.Injectable
+import com.keylesspalace.tusky.di.ViewModelFactory
+import com.keylesspalace.tusky.interfaces.LinkListener
+import com.keylesspalace.tusky.util.NetworkState
+import com.keylesspalace.tusky.util.Status
+import com.keylesspalace.tusky.util.hide
+import com.keylesspalace.tusky.util.show
+import kotlinx.android.synthetic.main.fragment_search.*
+import javax.inject.Inject
+
+abstract class SearchFragment<T> : Fragment(), LinkListener, Injectable {
+    private var snackbarErrorRetry: Snackbar? = null
+    @Inject
+    lateinit var viewModelFactory: ViewModelFactory
+
+    protected lateinit var viewModel: SearchViewModel
+
+    abstract fun createAdapter(): PagedListAdapter<T, *>
+
+    abstract val networkStateRefresh: LiveData<NetworkState>
+    abstract val networkState: LiveData<NetworkState>
+    abstract val data: LiveData<PagedList<T>>
+    protected lateinit var adapter: PagedListAdapter<T, *>
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        viewModel = ViewModelProviders.of(requireActivity(), viewModelFactory)[SearchViewModel::class.java]
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        return inflater.inflate(R.layout.fragment_search, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        initAdapter()
+        subscribeObservables()
+    }
+
+    private fun subscribeObservables() {
+        data.observe(viewLifecycleOwner, Observer {
+            adapter.submitList(it)
+        })
+
+        networkStateRefresh.observe(viewLifecycleOwner, Observer {
+            if (it == NetworkState.LOADING) {
+                searchProgressBar.show()
+            } else
+                searchProgressBar.hide()
```

Comment:
there is a extension function `visible` so you can do these checks shorter
`searchProgressBar.visible(it == NetworkState.LOADING)`